### PR TITLE
feat(show page): in progress.

### PR DESCRIPTION
**Pull request by DiLiUA**
- added responsive markup;
  - created timer component (only markup);
  - created shared component (only markup);
  - changed purchase component;
  - added query parameters for navigate to event page;

### CODE REVIEW

File: src/shared/share-container/share-container.component.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-share-container',
+  templateUrl: './share-container.component.html',
+  styleUrls: ['./share-container.component.scss']
+})
+export class ShareContainerComponent {
+
+  constructor() { }
```
Comment (Ugmaxie) [inline src/shared/share-container/share-container.component.ts]: drop empty constructor

File: src/shared/timer/timer.component.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+@Component({
+  selector: 'app-timer',
+  templateUrl: './timer.component.html',
+  styleUrls: ['./timer.component.scss']
+})
+export class TimerComponent {
+  @Input() datePerformance: string;
+
+  constructor() { }
```
Comment (Ugmaxie) [inline src/shared/timer/timer.component.ts]: drop empty constructor

File: src/show-page/show-page.component.html [hunk 3]
```diff
@@ -20,66 +20,18 @@
 
           <app-statistics [statistics]="currentShow?.statistics" [id]="currentShow?._id"></app-statistics>
```
Comment (Ugmaxie) [inline src/show-page/show-page.component.html:21]: if ID is string - don't use dual binding

File: src/show-page/show-page.component.html [hunk 4]
```diff
@@ -20,66 +20,18 @@
...
-              <div class="item-container">Hours</div>
-            </div>
-            <div class="dbl-dots">:</div>
-            <div class="minutes">
-              <div class="number-container">08</div>
-              <div class="item-container">Minutes</div>
-            </div>
-          </div>
-        </div>
+        <app-timer [datePerformance]="currentShow?.datePerformance"></app-timer>
```
Comment (Ugmaxie) [inline src/show-page/show-page.component.html:23]: if datePerformance is a string - don't use dual binding

File: src/show-page/show-page.component.ts [hunk 5]
```diff
@@ -1,16 +1,19 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
-import { Router } from '@angular/router';
+import { Router, ActivatedRoute } from '@angular/router';
+import { DomSanitizer } from '@angular/platform-browser';
+import { SafeHtml } from '@angular/platform-browser/src/security/dom_sanitization_service';
 import { Subscription } from 'rxjs/Subscription';
-import { head } from 'lodash';
 
+import { head } from 'lodash';
```
Comment (Ugmaxie) [inline src/show-page/show-page.component.ts]: lodash-es?

File: src/show-page/show-page.component.ts [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
 import * as moment from 'moment';
```
Comment (Ugmaxie) [inline src/show-page/show-page.component.ts]: if you use a simple moment methods which can be easily removed by native Date methods - don't use moment, it takes too much space in bundles.

File: src/show-page/show-page.component.ts [hunk 7]
```diff
@@ -19,44 +22,35 @@ import { Show } from '../event-launch/event-launch.model';
...
-
-    try {
-      const getCurrentShowFromLocalStorage = JSON.parse(this.localStorageService.getItem('currentShow'));
-      this.getCurrentShow(getCurrentShowFromLocalStorage);
-    } catch (err) {
-      console.error('something went wrong: ', err);
-    }
+    this.route.queryParams
+      .subscribe(params => {
+        console.log(params);
```
Comment (Ugmaxie) [inline src/show-page/show-page.component.ts]: remove console

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

File: src/show-page/show-page.component.html (hunk 3 above)
Comment (DiLiUA) [reply src/show-page/show-page.component.html:21]: This is not dual binding. It's `Input`

File: src/show-page/show-page.component.html (hunk 4 above)
Comment (DiLiUA) [reply src/show-page/show-page.component.html:23]: This is not dual binding. It's Input

File: src/artist-profile/tips/tips.component.ts [hunk 8]
```diff
@@ -1,5 +1,6 @@
 import { Component, Input, OnInit } from '@angular/core';
-import { forEach } from 'lodash';
+
+import { forEach } from 'lodash-es';
```
Comment (Ugmaxie) [inline src/artist-profile/tips/tips.component.ts]: import forEach from 'lodash-es/forEach';

File: src/shared/genres-filter/genres-filter.pipe.ts [hunk 9]
```diff
@@ -1,5 +1,6 @@
 import { Pipe, PipeTransform } from '@angular/core';
-import { filter } from 'lodash';
+
+import { filter } from 'lodash-es';
```
Comment (Ugmaxie) [inline src/shared/genres-filter/genres-filter.pipe.ts]: the same as forEach

File: src/shared/tips-container/tips-container.component.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  templateUrl: './tips-container.component.html',
+  styleUrls: ['./tips-container.component.scss']
+})
+export class TipsContainerComponent implements OnInit {
+  donation = 2;
+  donates: number[] = [2, 5, 7, 10, 15, 20, 30, 50];
+  donationPeriod = 'Month';
+  donationPeriodList: string[] = ['Year', 'Month', 'Week', 'Day', 'Once'];
+
+  constructor() { }
```
Comment (Ugmaxie) [inline src/shared/tips-container/tips-container.component.ts]: remove empty constructor

File: src/shared/tips-container/tips-container.component.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+  ngOnInit() {
```
Comment (Ugmaxie) [inline src/shared/tips-container/tips-container.component.ts]: remove empty ngOnInit

File: src/show-page/show-page.component.ts [hunk 12]
```diff
@@ -1,16 +1,18 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
-import { Router } from '@angular/router';
+import { Router, ActivatedRoute } from '@angular/router';
+import { DomSanitizer } from '@angular/platform-browser';
+import { SafeHtml } from '@angular/platform-browser/src/security/dom_sanitization_service';
 import { Subscription } from 'rxjs/Subscription';
-import { head } from 'lodash';
 
-import * as moment from 'moment';
+import { head } from 'lodash-es';
```
Comment (Ugmaxie) [inline src/show-page/show-page.component.ts]: the same as forEach

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

**Review by Ugmaxie — APPROVED** [review]

**Pull request merged**